src/Connector.js

```diff
@@ -37,15 +39,24 @@ class Connector {
   }
 
   async listenToCommands() {
-    await this.client.on('getData', async (cmd) => {
-      const { id, sensorIds } = cmd;
+    const { devices = [] } = await this.client.getDevices();
+    this.devices = devices
+      .filter((device) => !!device.schema)
+      .map((device) => device.id);
+
+    await Promise.all(this.devices.map((device) => this.registerListeners(device)));
+  }
+
+  async registerListeners(thingId) {
+    await this.client.on(`device.${thingId}.data.request`, async (msg) => {
+      const { id, sensorIds } = msg;
       this.onDataRequestedCb(id, sensorIds);
-    });
+    }, { consumerTag: `${thingId}-request` });
```

Comment:
The SDK users shouldn't worry about this communication detail. 👀

Reply:
Usually it doesn't. This is an omitted parameter that we use to manager consumer tags internally, You think is it a problem to use it here?

Reply:
As we discussed before, yes. It's a problem because we're violating the SDK purpose.